pkg/xds/generator/inbound_proxy_generator.go

```diff
@@ -36,11 +36,32 @@ func (g InboundProxyGenerator) Generate(ctx xds_context.Context, proxy *core_xds
 		protocol := core_mesh.ParseProtocol(iface.GetProtocol())
 
 		// generate CDS resource
-		localClusterName := envoy_names.GetLocalClusterName(endpoint.WorkloadPort)
-		clusterBuilder := envoy_clusters.NewClusterBuilder(proxy.APIVersion).
-			Configure(envoy_clusters.ProvidedEndpointCluster(localClusterName, false, core_xds.Endpoint{Target: endpoint.WorkloadIP, Port: endpoint.WorkloadPort})).
-			Configure(envoy_clusters.Timeout(defaults_mesh.DefaultInboundTimeout(), protocol))
-
+		var inboundListenerName string
+		var localClusterName string
+		var listenerPort uint32
+		var clusterBuilder *envoy_clusters.ClusterBuilder
+		if ctx.ControlPlane.EnableInboundPassthrough &&
+			proxy.Dataplane.Spec.IsUsingInboundTransparentProxy() &&
+			endpoint.WorkloadIP == "" {
+			listenerPort = endpoint.WorkloadPort
+			inboundListenerName = envoy_names.GetInboundListenerName(endpoint.DataplaneIP, endpoint.WorkloadPort)
+			localClusterName = envoy_names.GetInboundClusterName(endpoint.WorkloadPort)
+			clusterBuilder = envoy_clusters.NewClusterBuilder(proxy.APIVersion).
+				Configure(envoy_clusters.PassThroughCluster(localClusterName)).
+				Configure(envoy_clusters.UpstreamBindConfig(inPassThroughIPv4, 0)).
+				Configure(envoy_clusters.DefaultTimeout()).
+				Configure(envoy_clusters.CleanupInterval(60))
+		} else {
+			listenerPort = endpoint.DataplanePort
+			inboundListenerName = envoy_names.GetInboundListenerName(endpoint.DataplaneIP, endpoint.DataplanePort)
+			localClusterName = envoy_names.GetInboundClusterName(endpoint.WorkloadPort)
```

Comment:
should this be:
```go
localClusterName = envoy_names.GetInboundClusterName(endpoint. DataplanePort)
```
because otherwise `localClusterName` is the same regardless `EnableInboundPassthrough`